src/main/kotlin/nl/avisi/structurizr/site/generatr/site/views/SoftwareSystemHomePage.kt

```diff
@@ -7,7 +7,11 @@ import nl.avisi.structurizr.site.generatr.site.model.SoftwareSystemHomePageViewM
 
 fun HTML.softwareSystemHomePage(viewModel: SoftwareSystemHomePageViewModel) {
     softwareSystemPage(viewModel) {
-        h1 { +"Description" }
-        p { +viewModel.description }
+        if (viewModel.section != null)
+            markdown(viewModel, viewModel.section)
+        else {
```

Comment:
Instead of this `if` in the view, we could also move this logic to the `SoftwareSystemHomePageViewModel`, by letting it expose only a `MarkdownViewModel`. The view model should then decide between returning the first documentation section (if available), or returning a default `MarkdownViewModel`, containing what's in the `else` branch of this `if`.

This way, the logic is easily testable, and the view can be kept as "dumb" as possible.

Reply:
Yeah, that looks better. Thanks and done!